src/dialogs/more-info/controls/more-info-input_timetable.ts

```diff
@@ -0,0 +1,186 @@
+import "@material/mwc-icon-button/mwc-icon-button";
+import "../../../components/ha-svg-icon";
+import "../../../components/ha-switch";
+import type { HaSwitch } from "../../../components/ha-switch";
+import "../../../components/paper-time-input";
+import { mdiDelete, mdiPlus } from "@mdi/js";
+import { HassEntity } from "home-assistant-js-websocket";
+import { BINARY_STATE_ON, BINARY_STATE_OFF } from "../../../common/const";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+import { HomeAssistant } from "../../../types";
+
+declare type Event = {
+  time: string;
+  state: string;
+};
+
+@customElement("more-info-input_timetable")
+class MoreInfoInputTimetable extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({
+    attribute: false,
+    hasChanged(newVal: HassEntity, oldVal: HassEntity) {
+      if (!newVal) {
+        return false;
+      }
+      return (
+        !oldVal ||
+        JSON.stringify(oldVal.attributes.timetable) !==
+          JSON.stringify(newVal.attributes.timetable)
+      );
+    },
+  })
+  public stateObj?: HassEntity;
+
+  private timetable: Array<Event> = [];
+
+  private disable_reconfig = false;
```

Comment:
This doesn't seem to do anything or am I missing something??

Reply:
disable_reconfig is used to avoid loops while rendering:
- there are hooks to change in an entry which generates a call to the backend to update the entity
- when an entity is deleted, the way the UI framework works is that it copies the value from the next entry to the previous one (and eventually deletes the last entry). This process generated many calls to the backend, and basically put the system in a loop.
- to avoid it, we ignore changes in the entries, while they are being rendered. We only care about changes when they are triggered manually by the user.
- disable_reconfig is the flag for implementing this logic.

Reply:
I had assumed something like this, but I cannot see any place where the variable is actually read to safeguard against a loop. It is just ever written to (at least in the coding of this PR).